plugins/outputs/iotdb/iotdb.go

```diff
@@ -0,0 +1,287 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package iotdb
+
+// iotdb.go
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"math"
+
+	// Register IoTDB go client
+	"github.com/apache/iotdb-client-go/client"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+// DO NOT REMOVE THE NEXT TWO LINES! This is required to embed the sampleConfig data.
+//go:embed sample.conf
+var sampleConfig string
+
+type IoTDB struct {
+	Host            string `toml:"host"`
+	Port            string `toml:"port"`
+	User            string `toml:"user"`
+	Password        string `toml:"password"`
+	Timeout         int    `toml:"timeout"`
+	ConvertUint64To string `toml:"convertUint64To"`
+	TimeStampUnit   string `toml:"timeStampUnit"`
+	TreateTagsAs    string `toml:"treateTagsAs"`
```

Comment:
See `sample.conf` comments

```suggestion
	Timeout         config.Duration    `toml:"timeout"`
	ConvertUint64To string `toml:"uint64_conversion"`
	TimeStampUnit   string `toml:"timestamp_precision"`
	TreateTagsAs    string `toml:"convert_tags_to"`
```

Reply:
I will fix them.